Shopify/hydrogen
# Use Vite in templates and examples

### WHY are these changes introduced?

We want to dogfood Vite before the stable release.

### WHAT is this pull request doing?

- Use Vite in skeleton template
- Use Vite in `--diff` examples
- Remove the previous `example/vite` project
- Add a new `example/classic-remix` project
- Soft merge `dev` and `build` commands so that we can run `h2 dev` instead of `h2 dev-vite`. This merge will need more work a different PR.
- Patch `preview` and `deploy` commands to make them work with both Vite and Classic. This will also require another PR for cleanup.


### HOW to test your changes?

Run `dev`, `build` and `preview` commands in skeleton and examples. It should all work 🤞
